File: trunk/quote/Projects/PluginModel/common/PrimaryPropeties.vb
Imports System.ComponentModel
Imports System.Reflection

Imports Model
Imports Model.Template

Namespace Common

    Public Class PrimaryProperties
        Inherits SaveableProperties

        Private _ID As Integer

        <Browsable(False)> _
        Public Property CommonCustomer As New Customer

        <Browsable(False)> _
        Public Property CommonRequestForQuoteNumber As String = ""

        <Browsable(False)> _
        Public Property CommonPartNumber As String = ""

        <Browsable(False)> _
        Public Property CommonCreatedDate As DateTime = Date.Now

        <Browsable(False)> _
        Public Property CommonLastModified As DateTime = Date.Now

        <Browsable(False)> _
        Public Property CommonInitials As String

        <Browsable(False)> _
        Public ReadOnly Property CommonId As Integer
            Get
                Return _ID
            End Get
        End Property

        Public Sub SetId(ByVal id As Integer)
            Me._ID = id
            SendEvents()
        End Sub

    End Class

End Namespace

File: quote/Projects/PluginModel/common/PrimaryPropeties.cs
namespace Model.Common
{
    using System;
    using System.ComponentModel;
    using System.Linq;
    
    using Model.Template;

    public class PrimaryProperties : SavableProperties
    {
        private int id;
        
        private DateTime commonCreatedDate;
        
        private DateTime commonLastModified;
        
        public PrimaryProperties()
        {
            this.commonCreatedDate = DateTime.Now;
            this.commonLastModified = DateTime.Now;
        }
        
        [Browsable(false)]
        public Customer CommonCustomer { get; set; }

        [Browsable(false)]
        public string CommonRequestForQuoteNumber { get; set; }

        [Browsable(false)]
        public string CommonPartNumber { get; set; }

        [Browsable(false)]
        public DateTime CommonCreatedDate 
        {
            get { return this.commonCreatedDate; }
            set { this.commonCreatedDate = value; }
        }

        [Browsable(false)]
        public DateTime CommonLastModified 
        { 
            get { return this.commonLastModified; }
            set { this.commonLastModified = value; }
        }

        [Browsable(false)]
        public string CommonInitials { get; set; }

        [Browsable(false)]
        public int CommonId 
        {
            get { return this.id; }
        }

        public void SetId(int id)
        {
            this.id = id;
            this.OnPropertyChanged();
        }
    }
}
